src/nvim/CMakeLists.txt

```diff
@@ -120,9 +120,14 @@ if(NOT MSVC)
   set_source_files_properties(
     ${CONV_SOURCES} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
   # gperf generates ANSI-C with incorrect linkage, ignore it.
-  set_source_files_properties(
-    eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline -Wno-conversion")
-
+  check_c_compiler_flag(-Wno-static-in-inline HAS_WNO_STATIC_IN_INLINE_FLAG)
```

Comment:
When someone looks at this in two years time it will look like "we need to keep this, as we support compilers that don't have `-Wno-static-in-inline`" which is not the case; recent (as in "not ancient") gcc compilers don't have this flag either but are still fine without this workaround. What we really are guarding against, and what we only need to keep this along for, is compiling with gcc <= 4.3. 

But anyway, isn't openbsd shipping a nine (9) year old gcc their way of saying "you are really supposed to use clang if you can"?